### DISCUSSION TITLE
Component: Root tag inside an @if doesn't work

### DISCUSSION THREAD

**Discussion by levu42** [livewire/livewire#7354] [7. Bugs (archived)] [answered]
### Livewire version

3.2.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. have a template `@if(foo) <div id=a></div> @else <div id=b></div> @endif`
2. Try using that component
3. See that the regex https://github.com/livewire/livewire/blob/main/src/Drawer/Utils.php#L14 breaks because it has the if marker HTML comments around the root element

### Additional information

Don't have the time right now to PR a fix / failing test, but maybe someone else has before I'll find that time :)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@levu42 yeah that's correct, you should always have one root element, wrapping all `@if` etc.

**Reply by levu42** [reply]
@joshhanley thanks for the quick answer! I didn't realize that this also means the blade file, not just what the blade file renders to :) This is something that worked in LW2, broke now in an app that I migrated to LW3.

On a related note, thinking about what I actually tried solving here is something similar to blade components' [`shouldRender`](https://laravel.com/docs/10.x/blade#rendering-components) function. Something like this isn't implemented right now in Livewire, is it?

**Reply by joshhanley** [reply]
@levu42 hmm interesting, I'm surprised that worked in V2. But in V3, the Livewire component javascript object gets attached to the element directly, which in V2 it was in a global component store. So that's probably why it doesn't work now.

Ah right! Can you give me some context about what your component is? I'm just curious to how you would use it 🙂

**Reply by levu42** [reply]
@joshhanley The app is something where you can belong to many teams - if you have more than one team, there's a switcher component in the sidebar where you can switch for which team you're viewing the current page. If you only belong to one team, there's no need for the component, so it shouldn't take up any screen space.

I could of course implement that logic in the sidebar (`@if(auth()->user()->teams()->count() > 1) ... @endif`) but it feels way tidier to have that logic live in the component itself. In Livewire2 I had the whole component wrapped in `@if ($this->teams->count() > 1) <div id="team-switcher"></div> @endif`, now I needed to add another empty div around that as a wrapper. Maybe that's just the best solution for what I'm doing? Or maybe I should just render it and hide it via CSS? I just thought of how you can use a `shouldRender` function in blade components and I don't remember reading about something like that in the Livewire docs.
